### PR TITLE
Fixed editor [WAIV-1801]

### PR DESCRIPTION
**Pull request by konstantin1390**
https://maxiproject.atlassian.net/browse/WAIV-1801

### CODE REVIEW

File: src/client/post/EditPost/EditPost.js [hunk 1]
```diff
@@ -119,14 +121,16 @@ class EditPost extends Component {
         .then(campaignData => this.setState({ campaign: { ...campaignData, fetched: true } }))
         .catch(error => console.log('Failed to get campaign data:', error));
     }
+    getObject('jyp-cryptoinvestarena')
+      .then(data => this.setState({ defaultObjects: [data], objPercentage : setObjPercents([data], this.state.objPercentage) }),
```
Comment (milklim) [inline src/client/post/EditPost/EditPost.js:125]: add 'defaultObjects' to initial state

File: src/client/post/EditPost/EditPost.js [hunk 2]
```diff
@@ -137,21 +141,21 @@ class EditPost extends Component {
...
+    return compact(uniqBy([forecastObject, ...objEntities.map(entity => entity.data.object)], 'id'));
   };
 
   async handleChangeContent(rawContent) {
+    const { defaultObjects } = this.state;
     const nextState = { content: toMarkdown(rawContent) };
     this.getLinkedObjects(rawContent).then(linkedObjects => {
       const isLinkedObjectsChanged = !isEqual(this.state.linkedObjects, linkedObjects);
+      const linkedList = isLinkedObjectsChanged ? linkedObjects : this.state.linkedObjects;
+      const postObjects = !isEmpty(defaultObjects) ? [...defaultObjects, ...linkedList] : linkedList;
```
Comment (milklim) [inline src/client/post/EditPost/EditPost.js:153]: you could omit isEmpty condition if you add 'defaultObjects' to init state

File: src/client/post/EditPost/EditPost.js [hunk 3]
```diff
@@ -137,21 +141,21 @@ class EditPost extends Component {
...
       if (isLinkedObjectsChanged) {
-        const objPercentage = setObjPercents(linkedObjects, this.state.objPercentage);
         nextState.linkedObjects = linkedObjects;
-        nextState.objPercentage = objPercentage;
+        nextState.objPercentage = setObjPercents(postObjects, this.state.objPercentage);
+      } else {
+        nextState.objPercentage = setObjPercents(postObjects, this.state.objPercentage);
```
Comment (milklim) [inline src/client/post/EditPost/EditPost.js:158]: why do you use 'else' branch? Can we omit it?

File: src/client/post/EditPost/EditPost.js (hunk 1 above)
Comment (milklim) [inline src/client/post/EditPost/EditPost.js:125]: apply 'getClientWObj' to response data

File: src/client/helpers/wObjInfluenceHelper.js [hunk 4]
```diff
@@ -22,12 +22,12 @@ export const setInitialObjPercents = (linkedObjects, percentage) => {
...
   const firstPercent = 100 - averagePercent * (len - 1);
   const objPercentage = {
-    [first.id]: {
+    [first.id || first.author_permlink]: {
       percent: firstPercent,
     },
   };
   rest.forEach(obj => {
-    objPercentage[obj.id] = { percent: averagePercent };
+    objPercentage[obj.id || first.author_permlink] = { percent: averagePercent };
```
Comment (milklim) [inline src/client/helpers/wObjInfluenceHelper.js:30]: You could use only 'id' if apply 'getClientWObj' to all linked objects

File: src/client/post/EditPost/EditPost.js [hunk 5]
```diff
@@ -240,10 +257,10 @@ class EditPost extends Component {
     const currDraft = this.props.draftPosts.find(d => d.draftId === this.props.draftId);
     const oldMetadata = currDraft && currDraft.jsonMetadata;
     const waivioData = {
-      wobjects: linkedObjects.map(obj => ({
+      wobjects: postObjects.map(obj => ({
         objectName: obj.name,
-        author_permlink: obj.id,
-        percent: objPercentage[obj.id].percent,
+        author_permlink: obj.id || obj.author_permlink,
+        percent: objPercentage[obj.id || obj.author_permlink].percent,
```
Comment (milklim) [inline src/client/post/EditPost/EditPost.js:263]: You could use only 'id' if apply 'getClientWObj' to all linked objects

File: src/client/post/EditPost/EditPost.js [hunk 6]
```diff
@@ -257,24 +274,24 @@ class EditPost extends Component {
...
   handleForecastChange = (nextForecastValues, isReset) => {
-    const { linkedObjects, forecastValues } = this.state;
+    const { linkedObjects, forecastValues, defaultObjects } = this.state;
     const { isValid, wobjData } = nextForecastValues;
     if (isReset) {
       this.setState(prevState => {
         const filteredObjects = prevState.linkedObjects.filter(
           obj => obj.id !== get(prevState.forecastValues, ['wobjData', 'author_permlink'], ''),
         );
+        const postObjects = !isEmpty(defaultObjects) ? [...defaultObjects, ...filteredObjects] : filteredObjects;
```
Comment (milklim) [inline src/client/post/EditPost/EditPost.js:290]: `         const postObjects = [...defaultObjects, ...filteredObjects] `

File: src/client/post/EditPost/EditPost.js [hunk 7]
```diff
@@ -289,10 +306,11 @@ class EditPost extends Component {
           '',
         );
         const nextLinkedObjects = [getClientWObj(serverObject, this.props.locale), ...filtered];
+        const postObjects = !isEmpty(defaultObjects) ? [...defaultObjects, ...nextLinkedObjects] : nextLinkedObjects;
```
Comment (milklim) [inline src/client/post/EditPost/EditPost.js:309]: ` const postObjects = [...defaultObjects, ...nextLinkedObjects] `

File: src/client/post/EditPost/EditPost.js [hunk 8]
```diff
@@ -224,6 +229,18 @@ class EditPost extends Component {
...
+    const postObjects = isSubmitPost ? [...defaultObjects, ...linkedObjects] : linkedObjects;
+
+    if (isSubmitPost) {
+      defaultObjects.forEach(obj => {
+          const objName = obj.name || obj.default_name;
+          postData.body += `\n[${objName}](${getObjectUrl(
+            obj.id || obj.author_permlink,
+          )})&nbsp;\n`;
+        },
+      );
```
Comment (milklim) [inline src/client/post/EditPost/EditPost.js:242]: move this logic to handleSubmit function

File: src/client/post/EditPost/EditPost.js [hunk 9]
```diff
@@ -198,7 +202,7 @@ class EditPost extends Component {
     setTimeout(() => this.handleObjectSelect(object), 1200);
   }
 
-  buildPost() {
+  buildPost(isSubmit) {
```
Comment (milklim) [inline src/client/post/EditPost/EditPost.js:205]: remove isSubmit argument

File: src/client/post/EditPost/EditPost.js [hunk 10]
```diff
@@ -224,6 +229,18 @@ class EditPost extends Component {
       draftId,
       ...settings,
     };
+    const isSubmitPost =  isSubmit && !isEmpty(defaultObjects);
```
Comment (milklim) [inline src/client/post/EditPost/EditPost.js:232]: remove

File: src/client/post/EditPost/EditPost.js [hunk 11]
```diff
@@ -224,6 +229,18 @@ class EditPost extends Component {
...
+    const postObjects = isSubmitPost ? [...defaultObjects, ...linkedObjects] : linkedObjects;
```
Comment (milklim) [inline src/client/post/EditPost/EditPost.js:233]: const postObjects = [...defaultObjects, ...linkedObjects];

File: src/client/post/EditPost/EditPost.js (hunk 1 above)
Comment (konstantin1390) [reply src/client/post/EditPost/EditPost.js]: +

File: src/client/helpers/wObjInfluenceHelper.js (hunk 4 above)
Comment (konstantin1390) [reply src/client/helpers/wObjInfluenceHelper.js:30]: ++

File: src/client/post/EditPost/EditPost.js (hunk 2 above)
Comment (konstantin1390) [reply src/client/post/EditPost/EditPost.js]: ++

File: src/client/post/EditPost/EditPost.js (hunk 3 above)
Comment (konstantin1390) [reply src/client/post/EditPost/EditPost.js]: ++

File: src/client/post/EditPost/EditPost.js (hunk 7 above)
Comment (konstantin1390) [reply src/client/post/EditPost/EditPost.js]: ++

File: src/client/post/EditPost/EditPost.js (hunk 6 above)
Comment (konstantin1390) [reply src/client/post/EditPost/EditPost.js]: ++

File: src/client/post/EditPost/EditPost.js (hunk 5 above)
Comment (konstantin1390) [reply src/client/post/EditPost/EditPost.js:272]: ++

File: src/client/post/EditPost/EditPost.js (hunk 9 above)
Comment (konstantin1390) [reply src/client/post/EditPost/EditPost.js:217]: ++

File: src/client/post/EditPost/EditPost.js (hunk 10 above)
Comment (konstantin1390) [reply src/client/post/EditPost/EditPost.js]: ++

File: src/client/post/EditPost/EditPost.js (hunk 11 above)
Comment (konstantin1390) [reply src/client/post/EditPost/EditPost.js:245]: ++

File: src/client/post/EditPost/EditPost.js (hunk 8 above)
Comment (konstantin1390) [reply src/client/post/EditPost/EditPost.js]: ++

File: src/client/post/EditPost/EditPost.js [hunk 12]
```diff
@@ -137,22 +154,22 @@ class EditPost extends Component {
...
     this.getLinkedObjects(rawContent).then(linkedObjects => {
       const isLinkedObjectsChanged = !isEqual(this.state.linkedObjects, linkedObjects);
+      const linkedList = isLinkedObjectsChanged ? linkedObjects : this.state.linkedObjects;
+      const postObjects = [...defaultObjects, ...linkedList];
       if (isLinkedObjectsChanged) {
-        const objPercentage = setObjPercents(linkedObjects, this.state.objPercentage);
         nextState.linkedObjects = linkedObjects;
-        nextState.objPercentage = objPercentage;
       }
+      nextState.objPercentage = setObjPercents(postObjects, this.state.objPercentage);
```
Comment (milklim) [inline src/client/post/EditPost/EditPost.js:172]: return to ` if (isLinkedObjectsChanged) { ...` condition block

**Pull request merged**